sqllogictest/src/parser.rs

```diff
@@ -137,6 +137,35 @@ pub enum Record {
     Injected(Injected),
 }
 
+/// Newtype that allows comparing a RegEx
```

Comment:
I am not a super fan of this, but I can't think of any better way to implement `PartialEq` in a more maintainable way.

Reply:
What about https://mcarton.github.io/rust-derivative/latest/cmp.html#compare-with

Reply:
That seems to have worked well 👍